docs/topics/con_shared-messaging-work-queue-design-tradeoffs.adoc

```diff
@@ -0,0 +1,17 @@
+[id='con_shared-messaging-work-queue-design-tradeoffs_{context}']
+
+= {name-mission-shared-messaging-work-queue} design tradeoffs
+
+.Design Tradeoffs
+[width="100%",options="header"]
+|====================================================================
+|Pros           |Cons
+a|
+* Modularity: due to the decoupling between the workers and the central component of your messaging infrastructure, you can swap out the central component depending on which functionalities you need it to provide.
+For example, using an interconnect router instead of the broker sacrifices message persistence while allowing faster processing time per message.
```

Comment:
I believe the word Interconnect here refers to a concrete part of Red Hat AMQ; it's not a generic term (to the best of my knowledge). Hence, it should at least be capitalized, "Interconnect", preferably "Red Hat AMQ Interconnect". (Router doesn't need capitalization.)

Reply:
the term is taken form a conversation with Justin Ross an engineer form AMQ. If _Red Hat AMQ Interconnect_ is a product name, I'd probably consider using a more general term, such as _router_.
Would that work? WDYT?

Reply:
According to what Justin said, to some extent they do:
IIUC, _Broker_ refers to a messaging server that can store the message and forward it to the worker and then obtains acknowledgment from said worker that the message has been delivered.
The advantage is that if the recipient worker dies before receiving the message, the broker can re-route the message to a different one. The downside is a longer proscessing time per message.

A _Router_ is a messaging server that routes the message it receives form sending worker to the receiving one. There is no persistent storage of messages available and there is no verification whether the message has been received. The upshot is that the processing time per message is shorter.

Not sure if this is complex enough but that's the main difference as far as I've had a chance to learn.

Reply:
So it's "message router" or "dispatch router" here, I guess. I think I'd prefer "message router", as it doesn't use yet another new word.

Reply:
thank you:) I updated the section to use the term _message router_.